pkg/actions/service.go

```diff
@@ -260,29 +349,53 @@ func (s *Service) saveRunManifestObjectStore(ctx context.Context, manifest RunMa
 	return s.Writer.OutputWrite(ctx, storageNamespace, runManifestPath, manifestReader, manifestSize)
 }
 
-func (s *Service) saveRunManifestDB(ctx context.Context, repositoryID graveler.RepositoryID, manifest RunManifest) error {
-	_, err := s.DB.Transact(ctx, func(tx db.Tx) (interface{}, error) {
-		// insert run information
-		run := manifest.Run
-		_, err := tx.Exec(`INSERT INTO actions_runs(repository_id, run_id, event_type, start_time, end_time, branch_id, source_ref, commit_id, passed)
-			VALUES ($1,$2,$3,$4,$5,$6,$7,$8,$9)`,
-			repositoryID, run.RunID, run.EventType, run.StartTime, run.EndTime, run.BranchID, run.SourceRef, run.CommitID, run.Passed)
+func (s *KVService) saveRunManifestDB(ctx context.Context, repositoryID graveler.RepositoryID, manifest RunManifest) error {
+	// insert run information
+	run := manifest.Run
+	err := s.storeRun(ctx, protoFromRunResult(&run), repositoryID.String())
+	if err != nil {
+		return err
+	}
+
+	// insert each task information
+	for i := range manifest.HooksRun {
```

Comment:
Shouldn't the `hook_runs` be stored before the `run` itself?
Otherwise we'll have partial information if we'll save only the `run` without the `hooks`.